File: Kotlin/insertionSort.kt
fun insertionSort(arr: Array<Int>) {
    for (i in 1 until arr.size) {
        val key = arr[i]
        var j = i - 1

        while (j >= 0 && arr[j] > key) {
            arr[j + 1] = arr[j]
            j -= 1
        }
        arr[j + 1] = key
    }
}

fun main() {
    val arr = arrayOf(12, 11, 13, 5, 6, 0, -1, 2)
    insertionSort(arr)
    println("Sorted array: ${arr.joinToString(", ")}")
}

File: PHP/insertion_sort.php
<?php
function insertionSort(array $arr): array {
    $n = count($arr);
    for ($i = 1; $i < $n; $i++) {
        $key = $arr[$i];
        $j = $i - 1;

        while ($j >= 0 && $arr[$j] > $key) {
            $arr[$j + 1] = $arr[$j];
            $j--;
        }
        $arr[$j + 1] = $key;
    }
    return $arr;
}

$unsortedArray = [64, 34, 25, 12, 22, 11, 90, -89, 0, -45];
$sortedArray = insertionSort($unsortedArray);

echo "Sorted Array: " . implode(", ", $sortedArray);
?>
